dart/lib/src/hub.dart

```diff
@@ -0,0 +1,280 @@
+import 'sentry_options.dart';
+
+typedef ScopeCallback = void Function(Scope);
+
+/// SDK API contract which combines a client and scope management
+class Hub {
+  static SentryClient _getClient({SentryOptions fromOptions}) {
+    return SentryClient(
+      dsn: fromOptions.dsn,
+      environmentAttributes: fromOptions.environmentAttributes,
+      compressPayload: fromOptions.compressPayload,
+      httpClient: fromOptions.httpClient,
+      clock: fromOptions.clock,
+      uuidGenerator: fromOptions.uuidGenerator,
+    );
+  }
+
+  final ListQueue<_StackItem> _stack;
+
+  final SentryOptions _options;
+
+  factory Hub(SentryOptions options) {
+    _validateOptions(options);
+
+    return Hub._(options);
+  }
+
+  Hub._(SentryOptions options)
+      : _options = options,
+        _stack = ListQueue() {
+    _stack.add(_StackItem(_getClient(fromOptions: options), Scope(_options)));
+    _isEnabled = true;
+  }
+
+  static void _validateOptions(SentryOptions options) {
+    if (options == null) {
+      throw ArgumentError.notNull('options');
+    }
+
+    if (options.dsn == null) {
+      throw ArgumentError.notNull('options.dsn');
+    }
+  }
+
+  bool _isEnabled = false;
+
+  /// Check if the Hub is enabled/active.
+  bool get isEnabled => _isEnabled;
+
+  SentryId _lastEventId;
+
+  /// Last event id recorded in the current scope
+  SentryId get lastEventId => _lastEventId;
+
+  /// Captures the event.
+  Future<SentryId> captureEvent(Event event) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureEvent' call is a no-op.",
+      );
+    } else if (event == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureEvent called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
+      if (item != null) {
+        try {
+          sentryId = await item.client.captureEvent(event, scope: item.scope);
+        } catch (err) {
+          /* TODO add Event.id */
+          _options.logger(
+            SeverityLevel.error,
+            'Error while capturing event with id: ${event}',
+          );
+        }
+      } else {
+        _options.logger(
+          SeverityLevel.fatal,
+          'Stack peek was null when captureEvent',
+        );
+      }
+    }
+    _lastEventId = sentryId;
+    return sentryId;
+  }
+
+  /// Captures the exception
+  Future<SentryId> captureException(
+    dynamic throwable, {
+    dynamic stackTrace,
+  }) async {
+    var sentryId = SentryId.empty();
+
+    if (!_isEnabled) {
+      _options.logger(
+        SeverityLevel.warning,
+        "Instance is disabled and this 'captureException' call is a no-op.",
+      );
+    } else if (throwable == null) {
+      _options.logger(
+        SeverityLevel.warning,
+        'captureException called with null parameter.',
+      );
+    } else {
+      final item = _stack.first;
```

Comment:
Maybe we should just add push and pop scope now, given that all the work to support is is done here

Reply:
we could add it later too, let's focus on the missing and required parts for now, of course, if some time left, we'll be back to it even on this sprint